dask/dataframe/core.py

```diff
@@ -1645,7 +1645,9 @@ def count(self, axis=None, split_every=False):
                 split_every=split_every,
             )
             if isinstance(self, DataFrame):
-                result.divisions = (min(self.columns), max(self.columns))
+                result.divisions = (self.columns.min(), self.columns.max())
+                if len(result) == 0:
```

Comment:
Why is this astype needed? Is this case tested?

Reply:
pandas result `dtype` is `int64`
```
>>> import dask.dataframe as dd
>>> import pandas as pd
>>> pdf = pd.DataFrame()
>>> ddf = dd.from_pandas(pdf,1)
>>> pdf.count()
Series([], dtype: int64)
>>> ddf.count().compute()
Series([], dtype: float64)
```

Reply:
it is tested by `test_empty_df_reductions("count")`

Reply:
because of aggregation where it sums up the counts. `sum` return `dtype` is `float64`

Reply:
Right, my question is about what it's float64. Are we creating an empty array somewhere, and so the dtype is inferred to be float rather than int?

Reply:
seems to be coming from here:
```
def _concat(args):
    if not args:
        return args
    if isinstance(first(core.flatten(args)), np.ndarray):
        return da.core.concatenate3(args)
    if not has_parallel_type(args[0]):
        try:
            return pd.Series(args)
        except Exception:
            return args
    # We filter out empty partitions here because pandas frequently has
    # inconsistent dtypes in results between empty and non-empty frames.
    # Ideally this would be handled locally for each operation, but in practice
    # this seems easier. TODO: don't do this.
    args2 = [i for i in args if len(i)]
    return args[0] if not args2 else methods.concat(args2, uniform=True)
```

Reply:
Mmm, I'm not sure this is the root problem, though I may be wrong. I can try taking a closer look later, but it may be a while.

Reply:
Sorry for the delay in this @exemplary-citizen. Still planning to dig into this, hopefully tomorrow.

Reply:
Sounds good!

Reply:
Fixed in https://github.com/dask/dask/pull/5804/commits/42681a0bc3fb03de344f87ded4517f3502ba99ea. The problem is that `pd.DataFrame(index=[0]).sum()` returns a float dtype series.

We need to avoid the `if len(result) == 0`, since that triggers a computation of the task graph up to this point.